store/mockstore/mocktikv/cluster.go

```diff
@@ -269,6 +270,49 @@ func (c *Cluster) GetRegionByID(regionID uint64) (*metapb.Region, *metapb.Peer)
 	return nil, nil
 }
 
+// ScanRegions returns at most `limit` regions from given `key` and their leaders.
+func (c *Cluster) ScanRegions(key []byte, limit int) ([]*metapb.Region, []*metapb.Peer) {
+	c.RLock()
+	defer c.RUnlock()
+
+	regions := make([]*Region, 0, len(c.regions))
+	for _, region := range c.regions {
+		regions = append(regions, region)
+	}
+
+	sort.Slice(regions, func(i, j int) bool {
+		return bytes.Compare(regions[i].Meta.GetStartKey(), regions[j].Meta.GetStartKey()) < 0
+	})
+
+	keyLocation := sort.Search(len(regions), func(i int) bool {
+		endKey := regions[i].Meta.GetEndKey()
+		if len(endKey) == 0 {
+			return true
+		}
+		return bytes.Compare(regions[i].Meta.GetEndKey(), key) > 0
+	})
+	regions = regions[keyLocation:]
+	if len(regions) > limit {
+		regions = regions[:limit]
+	}
+
+	metas := make([]*metapb.Region, 0, len(regions))
+	leaders := make([]*metapb.Peer, 0, len(regions))
+	for _, region := range regions {
+		leader := region.leaderPeer()
+		if leader == nil {
+			leader = &metapb.Peer{}
```

Comment:
Seems violate the semantics of this function if we return invalid leader.

Reply:
This is the same as PD's ScanRegions API. It uses a repeated field for leaders so that it can't be nil. Therefore we judge if a region has no leader by checking if the leader's ID is 0.